### PR TITLE
ci: add cluster exclusion list to cleanup

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
Adds an exclusion list to our cleanup script to keep clusters around for investigation purposes. Motivated by ongoing SQS failures in our soak tests.

**How was this change tested?**
N/A

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/hack/resource/clean/main.go [hunk 1]
```diff
@@ -26,13 +26,18 @@ import (
...
+	"golang.org/x/exp/slices"
 
 	"github.com/aws/karpenter-provider-aws/test/hack/resource/pkg/metrics"
 	"github.com/aws/karpenter-provider-aws/test/hack/resource/pkg/resourcetypes"
 )
 
 const sweeperCleanedResourcesTableName = "sweeperCleanedResources"
 
+var excludedClusters = []string{
+	"soak-periodic-46287782",
```
Comment (jonathan-innis) [inline test/hack/resource/clean/main.go:38]: Could this be a tag rather than doing this by name?

Comment (jonathan-innis) [inline test/hack/resource/clean/main.go:38]: TODO statement to remove this from the file once we are done with the investigation

Comment (jonathan-innis) [reply test/hack/resource/clean/main.go:38]: We'd have to go tag every resource individually which is a lot of work and a lot to remember. After discussion offline, I agree that checking the cluster name and the tag that we are targeting is the best way to handle this

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — DISMISSED** [review]

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Pull request merged**